backend/demo/src/main/java/app/k9mail/backend/demo/DemoBackend.kt

```diff
@@ -0,0 +1,208 @@
+        throw UnsupportedOperationException("not implemented")
+    }
+
+    override fun setFlag(folderServerId: String, messageServerIds: List<String>, flag: Flag, newState: Boolean) = Unit
+
+    override fun markAllAsRead(folderServerId: String) = Unit
+
+    override fun expunge(folderServerId: String) {
+        throw UnsupportedOperationException("not implemented")
+    }
+
+    override fun expungeMessages(folderServerId: String, messageServerIds: List<String>) {
+        throw UnsupportedOperationException("not implemented")
+    }
+
+    override fun deleteMessages(folderServerId: String, messageServerIds: List<String>) = Unit
+
+    override fun deleteAllMessages(folderServerId: String) = Unit
+
+    override fun moveMessages(
+        sourceFolderServerId: String,
+        targetFolderServerId: String,
+        messageServerIds: List<String>
+    ): Map<String, String> {
+        return messageServerIds.associateWith { randomUuid() }
+    }
+
+    override fun moveMessagesAndMarkAsRead(
+        sourceFolderServerId: String,
+        targetFolderServerId: String,
+        messageServerIds: List<String>
+    ): Map<String, String> {
+        return messageServerIds.associateWith { randomUuid() }
+    }
+
+    override fun copyMessages(
+        sourceFolderServerId: String,
+        targetFolderServerId: String,
+        messageServerIds: List<String>
+    ): Map<String, String> {
+        return messageServerIds.associateWith { randomUuid() }
+    }
+
+    override fun search(
+        folderServerId: String,
+        query: String?,
+        requiredFlags: Set<Flag>?,
+        forbiddenFlags: Set<Flag>?,
+        performFullTextSearch: Boolean
+    ): List<String> {
+        throw UnsupportedOperationException("not implemented")
+    }
+
+    override fun fetchPart(folderServerId: String, messageServerId: String, part: Part, bodyFactory: BodyFactory) {
+        throw UnsupportedOperationException("not implemented")
+    }
+
+    override fun findByMessageId(folderServerId: String, messageId: String): String? {
+        throw UnsupportedOperationException("not implemented")
+    }
+
+    override fun uploadMessage(folderServerId: String, message: Message): String {
+        return randomUuid()
+    }
+
+    override fun checkIncomingServerSettings() = Unit
+
+    override fun checkOutgoingServerSettings() = Unit
+
+    override fun sendMessage(message: Message) {
+        val inboxServerId = messageStoreInfo.filterValues { it.type == FolderType.INBOX }.keys.first()
+        val backendFolder = backendStorage.getFolder(inboxServerId)
+
+        val newMessage = message.copy(uid = randomUuid())
+        backendFolder.saveMessage(newMessage, MessageDownloadState.FULL)
+    }
+
+    override fun createPusher(callback: BackendPusherCallback): BackendPusher {
+        throw UnsupportedOperationException("not implemented")
+    }
+
+    private fun randomUuid() = UUID.randomUUID().toString()
```

Comment:
Note to self: Rename to `createNewServerId()`? The fact that it's a random UUID is an irrelevant implementation detail.